### PR TITLE
Option to transform text before rendering

### PR DESCRIPTION
**Pull request by menion**
New tag that allows to transform text before rendering.

**Tag**: "text-transform"

**Options**:
- "none" > default, no change
- "uppercase" > all letters uppercase
- "lowercase" > all letters lowercase
- "capitalize" > first letter capitalized

**Usable for**:
- area caption
- POI caption
- way text

### CODE REVIEW

File: mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/RenderInstruction.java [hunk 1]
```diff
@@ -173,4 +175,22 @@ protected Scale scaleFromValue(String value) {
...
+    protected String transformText(String text, TextTransform transformation) {
+        if (text.length() == 0) {
+            return text;
+        }
+        if (transformation == TextTransform.UPPERCASE) {
+            return text.toUpperCase();
+        } else if (transformation == TextTransform.LOWERCASE) {
+            return text.toLowerCase();
+        } else if (transformation == TextTransform.CAPITALIZE) {
+            return text.substring(0, 1).toUpperCase() + text.substring(1);
```
Comment (devemux86) [inline mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/RenderInstruction.java:191]: The case conversions are to be done with `Locale.getDefault()`?

Comment (menion) [reply mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/RenderInstruction.java:191]: Yes, that's right. Do you think there is a better option?

Comment (devemux86) [reply mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/RenderInstruction.java:191]: Some recommend `Locale.ROOT` in generic case conversions,
however these are local names, so not sure what is best option.

Comment (menion) [reply mapsforge-map/src/main/java/org/mapsforge/map/rendertheme/renderinstruction/RenderInstruction.java]: Oki, thanks. Let's give it a try.

**Comment by devemux86** [comment]
@menion thanks!

**Comment by devemux86** [comment]
See https://github.com/mapsforge/vtm/issues/991:
- Implement in VTM themes
- Support Mapsforge themes

**Pull request merged**